subsys/dfu/dfu_target/src/dfu_target_modem_delta.c

```diff
@@ -33,7 +33,7 @@ static int delete_banked_modem_delta_fw(void)
 
 	LOG_INF("Deleting firmware image, this can take several minutes");
 	err = nrf_modem_delta_dfu_erase();
-	if (err != 0) {
+	if (err != 0 && err != NRF_MODEM_DELTA_DFU_ERASE_PENDING) {
```

Comment:
Why do we need to accept the ERASE_PENDING error?

Reply:
Because `reset()` is a non blocking, so it might still be erasing when someone call init again.

So either erase, or wait for previous erase operation to complete.